### PR TITLE
feat: add the ability to read RNTuple alias columns

### LINKED ISSUE #928: Add support for current RNTuple files
Due to some recent changes in RNTuple's format, uproot is not able to read RNTuples correctly anymore.

A reproducer:

```
python -c 'import uproot; print(uproot.__version__); uproot.open("https://xrootd-local.unl.edu:1094//store/user/AGC/nanoaod-rntuple/zstd/TT_TuneCUETP8M1_13TeV-powheg-pythia8/cmsopendata2015_ttbar_19980_PU25nsData2015v1_76X_mcRun2_asymptotic_v12_ext3-v1_00000_0000.root")["Events"].arrays(["nTau"])
```
results in

```
5.0.10
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/blue/Repos/analysis-grand-challenge-iris-hep/analyses/cms-open-data-ttbar/venv-uproot5/lib/python3.11/site-packages/uproot/models/RNTuple.py", line 394, in arrays
    entry_stop = entry_stop or self._length
                               ^^^^^^^^^^^^
  File "/home/blue/Repos/analysis-grand-challenge-iris-hep/analyses/cms-open-data-ttbar/venv-uproot5/lib/python3.11/site-packages/uproot/models/RNTuple.py", line 180, in _length
    return sum(x.num_entries for x in self.cluster_summaries)
                                      ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/blue/Repos/analysis-grand-challenge-iris-hep/analyses/cms-open-data-ttbar/venv-uproot5/lib/python3.11/site-packages/uproot/models/RNTuple.py", line 175, in cluster_summaries
    return self.footer.cluster_summaries
           ^^^^^^^^^^^
  File "/home/blue/Repos/analysis-grand-challenge-iris-hep/analyses/cms-open-data-ttbar/venv-uproot5/lib/python3.11/site-packages/uproot/models/RNTuple.py", line 164, in footer
    f = FooterReader().read(self._footer_chunk, cursor, context)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/blue/Repos/analysis-grand-challenge-iris-hep/analyses/cms-open-data-ttbar/venv-uproot5/lib/python3.11/site-packages/uproot/models/RNTuple.py", line 697, in read
    out.extension_links = self.extension_header_links.read(chunk, cursor, context)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/blue/Repos/analysis-grand-challenge-iris-hep/analyses/cms-open-data-ttbar/venv-uproot5/lib/python3.11/site-packages/uproot/models/RNTuple.py", line 556, in read
    assert num_bytes < 0, f"num_bytes={num_bytes}"
           ^^^^^^^^^^^^^
AssertionError: num_bytes=36
```

IIUC the Julia implementation already implemented the necessary changes at https://github.com/JuliaHEP/UnROOT.jl/pull/264/files

### PR DESCRIPTION
**Pull request by ioanaif**
Fixes #928 and #975

### CODE REVIEW

**Comment by jpivarski** [comment]
The new **docs/readthedocs.org:uproot** test requirement can be satisfied by merging #1084/[jpivarski/fix-readthedocs-documentation](https://github.com/scikit-hep/uproot5/tree/jpivarski/fix-readthedocs-documentation) into this PR (or by merging in `main`, which has it).

**Review by jpivarski — APPROVED** [review]
This PR is a big step forward for handling alias columns. I saw it in many stages of development, and the final diff is small but a definite improvement. The new tests have to be marked `xfail`, but that will be a good starting point for @ariostas.

File: src/uproot/models/RNTuple.py [hunk 1]
```diff
@@ -457,17 +456,56 @@ def arrays(
             [c.num_entries for c in clusters[start_cluster_idx:stop_cluster_idx]]
         )
 
+        self._alias_columns_dict = {
+            el.field_id: el.physical_id
+            for i, el in enumerate(self.header.alias_columns)
+        }
+        self._column_recorrds_dict = {}
```
Comment (ariostas) [inline src/uproot/models/RNTuple.py:463]: Very minor comment: recorrds -> records?

Comment (jpivarski) [reply src/uproot/models/RNTuple.py:463]: You're right, thanks! See #1088.

**Pull request merged**